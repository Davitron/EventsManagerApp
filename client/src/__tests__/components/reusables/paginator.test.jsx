import React from 'react';
import expect from 'expect';
import { shallow } from 'enzyme';
import Paginator from '../../../components/reusables/Paginator';

let mountedComponent;
let props;
let wrapper;

const pagingData = {
  limit: 10,
  offset: 0,
  page: 1,
  pages: 5,
  count: 50,
  currentPageSize: 10
};

const onChange = () => null;
const onShowSizeChange = () => null;

const getComponent = () => {
  if (!mountedComponent) {
    props = {
      pagingData,
      onChange,
      onShowSizeChange
    };
    mountedComponent = shallow(<Paginator {...props} />);
  }
  return mountedComponent;
};

describe('Paginator Component', () => {
  it('should successfully render the component', () => {
    wrapper = getComponent();
    expect(wrapper).toMatchSnapshot();
  });
});
