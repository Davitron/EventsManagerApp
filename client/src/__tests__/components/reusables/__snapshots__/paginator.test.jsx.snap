// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Paginator Component should successfully render the component 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Paginator
    onChange={[Function]}
    onShowSizeChange={[Function]}
    pagingData={
      Object {
        "count": 50,
        "currentPageSize": 10,
        "limit": 10,
        "offset": 0,
        "page": 1,
        "pages": 5,
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "className": "pagination",
      "current": 1,
      "defaultCurrent": 1,
      "defaultPageSize": 10,
      "hideOnSinglePage": false,
      "itemRender": [Function],
      "locale": Object {
        "items_per_page": "Items",
      },
      "onChange": [Function],
      "onShowSizeChange": [Function],
      "pageSize": 10,
      "pageSizeOptions": Array [
        "9",
        "12",
        "18",
      ],
      "prefixCls": "rc-pagination",
      "selectComponentClass": [Function],
      "selectPrefixCls": "rc-select",
      "showLessItems": false,
      "showPrevNextJumpers": true,
      "showQuickJumper": false,
      "showSizeChanger": true,
      "showTitle": true,
      "style": Object {},
      "total": 50,
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "className": "pagination",
        "current": 1,
        "defaultCurrent": 1,
        "defaultPageSize": 10,
        "hideOnSinglePage": false,
        "itemRender": [Function],
        "locale": Object {
          "items_per_page": "Items",
        },
        "onChange": [Function],
        "onShowSizeChange": [Function],
        "pageSize": 10,
        "pageSizeOptions": Array [
          "9",
          "12",
          "18",
        ],
        "prefixCls": "rc-pagination",
        "selectComponentClass": [Function],
        "selectPrefixCls": "rc-select",
        "showLessItems": false,
        "showPrevNextJumpers": true,
        "showQuickJumper": false,
        "showSizeChanger": true,
        "showTitle": true,
        "style": Object {},
        "total": 50,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
